src/main/java/in/projecteka/gateway/common/Validator.java

```diff
@@ -27,24 +26,30 @@
     CacheAdapter<String, String> requestIdMappings;
 
 
-    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity) {
-        List<String> xHipIds = requestEntity.getHeaders().get(X_HIP_ID);
-        if (xHipIds == null || xHipIds.isEmpty()) {
-            return Mono.error(ClientError.hipIdMissing());
+    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity, String id) {
+        List<String> ids = requestEntity.getHeaders().get(id);
```

Comment:
You can use this method https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpHeaders.html#getFirst-java.lang.String-

if you are looking for only one header value, it returns null if header does not exists.

then you can use StringUtils from springframework for checking hasText =>  https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StringUtils.html#hasText-java.lang.String-